tests/e2e/e2e_workflow.spec.js

```diff
@@ -36,20 +37,35 @@ describe('Basic end-to-end Workflow', function () {
     testArgs.push('Chrome_travis_ci');
   }
 
+
+  // We don't want to use npm link because then npm dependencies
+  // that only exist in the project will not be accessible to CLI
+  // This is particularly problematic for --mobile, which uses Universal
+  // libs as part of the build process.
+  // Instead, we'll pack CLI as a tarball
   it('Installs angular-cli correctly', function () {
     this.timeout(300000);
 
-    sh.exec('npm link', { silent: true });
+    sh.exec('npm pack', { silent: true });
+    expect(existsSync(path.join(process.cwd(), `angular-cli-${repoPkgJson.version}.tgz`)));
     return tmp.setup('./tmp').then(function () {
       process.chdir('./tmp');
-      expect(existsSync(path.join(process.cwd(), 'bin', 'ng')));
     });
   });
 
+
   it('Can create new project using `ng new test-project`', function () {
     this.timeout(4200000);
-
-    return ng(['new', 'test-project', '--link-cli=true']).then(function () {
+    let args = ['--skip-npm'];
+    // If testing in the mobile matrix on Travis, create project with mobile flag
+    if (isMobileTest()) {
+      args = args.concat(['--mobile']);
+    }
+    return ng(['new', 'test-project'].concat(args)).then(function () {
+      // Install Angular CLI from packed version
+      let tarball = path.resolve(root, `../angular-cli-${repoPkgJson.version}.tgz`);
+      sh.exec(`npm install && npm install ${tarball}`);
```

Comment:
Discussed in person: there is added latency during install, since it's running an additional `npm install` to install the tarball.